If built for Cygwin itself, not only is it unnecessary to use cygpath,
but with Cygwin 1.7, ">NUL" is understood as a redirect to file "NUL"
instead of the Windows null device (equiv. /dev/null).

--- origsrc/flexdll/reloc.ml	2009-08-07 01:45:20.611397000 -0500
+++ src/flexdll/reloc.ml	2009-08-07 01:52:12.639945100 -0500
@@ -1032,11 +1032,10 @@ let main () =
 
   use_cygpath :=
     begin
-      match !toolchain, !cygpath_arg with
-      | _, `Yes -> true
-      | _, `No -> false
-      | (`CYGWIN|`MINGW), `None -> (Sys.command "cygpath -v 2>NUL >NUL" = 0)
-      | (`MSVC|`LIGHTLD), `None -> false
+      match Sys.os_type, !toolchain, !cygpath_arg with
+      | _, _, `Yes -> true
+      | "Win32", (`CYGWIN|`MINGW), `None -> (Sys.command "cygpath -v 2>NUL >NUL" = 0)
+      | _, _, _ -> false
     end;
 
 
